packages/rps/src/containers/GameContainer.tsx

```diff
@@ -65,64 +71,73 @@ function RenderGame(props: GameProps) {
       return (
         <WeaponSelectedPage
           message="Waiting for your opponent to choose their move"
-          yourWeapon={state.myWeapon}
+          yourWeapon={localState.myWeapon}
         />
       );
     case 'ResultPlayAgain':
       return (
         <WaitForResting
-          yourWeapon={state.myWeapon}
-          theirWeapon={state.theirWeapon}
-          result={state.result}
+          yourWeapon={localState.myWeapon}
+          theirWeapon={localState.theirWeapon}
+          result={localState.result}
           playAgain={props.playAgain}
           waitForOpponent={false}
         />
       );
     case 'WaitForRestart':
       return (
         <WaitForResting
-          yourWeapon={state.myWeapon}
-          theirWeapon={state.theirWeapon}
-          result={state.result}
+          yourWeapon={localState.myWeapon}
+          theirWeapon={localState.theirWeapon}
+          result={localState.result}
           playAgain={props.playAgain}
           waitForOpponent={true}
         />
       );
-    case 'ShuttingDown':
+    case 'InsufficientFunds':
       return (
-        <ResultPage
-          yourWeapon={state.myWeapon}
-          theirWeapon={state.theirWeapon}
-          result={state.result}
-          playAgain={props.playAgain}
-          shutDownReason={state.reason}
+        <InsufficientFunds
+          yourWeapon={localState.myWeapon}
+          theirWeapon={localState.theirWeapon}
+          result={localState.result}
+          action={props.gameOver}
         />
       );
+    case 'Resigned':
+      const ourTurn =
+        localState.player === 'A'
+          ? Number(channelState.turnNum) % 2 !== 0
+          : Number(channelState.turnNum) % 2 === 0;
+      const iResigned =
+        (ourTurn && channelState.status === 'running') ||
+        (!ourTurn && channelState.status === 'closing') ||
+        (ourTurn && channelState.status === 'closed');
```

Comment:
I feel like this should be stored in the state. It seems like too much logic to be in the view layer.

Reply:
Addressed in 4d98af6